impl/maven-impl/src/main/java/org/apache/maven/impl/model/DefaultModelProcessor.java

```diff
@@ -69,25 +72,31 @@
 public class DefaultModelProcessor implements ModelProcessor {
 
     private final ModelXmlFactory modelXmlFactory;
-    private final List<ModelParser> modelParsers;
+    private final @Nullable List<ModelParser> modelParsers;
 
-    @Inject
```

Comment:
its not CDI but dont needed like in spring as its working without (convention over config)

Reply:
Are you aware that Maven has its own DI framework ? If you remove the required annotations, it will probably break ?
@Nullable can be used to inform the DI that the requirement is optional.
Did you run UTs and ITs ?

Reply:
> Did you run UTs and ITs ?

always run root `clean install`

> Are you aware that Maven has its own DI framework ? If you remove the required annotations, it will probably break ?

i delete/challenge everything and see if it breaks. All worked out locally thats why i want to lern and clarify, thx.

Reply:
<img width="729" alt="image" src="https://github.com/user-attachments/assets/529d9cd3-e9b7-4926-b576-36419ed6009c" />